src/classes/RD2_RecurringDonation.cls

```diff
@@ -245,7 +245,25 @@ public inherited sharing class RD2_RecurringDonation {
     * @return Boolean
     */
     public Boolean isFixedLength() {
-        return rd.RecurringType__c == RD2_Constants.RECURRING_TYPE_FIXED;
+        if (!RD2_EnablementService.isRecurringDonations2EnabledOverride) {
+            return rd.RecurringType__c == RD2_Constants.RECURRING_TYPE_FIXED;
+        }
+
+        if (UTIL_Describe.isValidPicklistOption(String.valueOf(
```

Comment:
To reduce performance implications, I recommend doing the check for the picklist value only once rather than for each individual RD record.

Reply:
@RadSFDeveloper @force2b 

I'd like to discuss this a bit with you and Radmila.  Since RD2_RecurringDonation.cls is a wrapper for the record, we cannot use this logic without moving it out of this class or converting it to a static method within this class.  I debated with myself back and forth on how to handle this and I am frankly not sure of the right choice. 

My initial take was the same as yours, Mike, that we should move it somewhere appropriate for a static method that is cached for the entire context.  But, I think that implies changing RD2_RecurringDonation.isFixedLength() to a static method as well, or at least a method that calls the static context.  Either that or we do not try to re-use the logic.

What do the two of you think about this?

Reply:
@force2b @RadSFDeveloper Also, we discussed the proxy idea in our team dev meeting yesterday before you arrived.  I had initially used npe03__Open_Ended_Status__c, but there were concerns that a 3rd party validation might rely on that.

Reply:
With regards to proxy field: That is right, changing an RD1 field might trigger client's validations or triggers that will prevent validation to succeed. Thus, the idea is to use the Open Ended Status as-is in addition to other checks to determine if the RD is a fixed-length. So, if it is a dry run and Recurring Type is Open but the Open Ended Status is None/`null`, then this is a simulated fixed-length RD. In other conditions, we will check for the Recurring Type Fixed as usual to determine if the RD is fixed-length.

With regards to performance: I agree we should be using a cache (static var/method) to check if Recurring Type "Fixed" value is present. I thought we will use a new static method in `RD2_EnablementService` for it (I have not reviewed the PR yet so perhaps it is already done). We can discuss details offline.